### ISSUE TITLE
Running Kaniko w/ options --no-push=true --cache=true and --cache-dir="..." leads to error 

### ISSUE THREAD

**Issue by stawr93** [GoogleContainerTools/kaniko#2456]
I'd like to use caching to directory without publishing layers to any repo/registry.

**Actual behavior**
Using `--no-push=true --cache=true --cache-dir=/some-path` options in combine leads to an error:
```
Error: cache flags invalid: if using cache with --no-push, specify cache repo with --cache-repo
```

**Expected behavior**
In case of `--no-push=true --cache=true --cache-dir=/some-path` Kaniko will cache layers into provided directory and will not push them into any repo.

**To Reproduce**
1. Run `/kaniko/executor --context "<context>" --dockerfile "<path>/Dockerfile" --destination "<some custom tag>" --no-push --cache=true --cache-dir=<some path>`

**Additional Information**
 - Kaniko Image (fully qualified with digest): `gcr.io/kaniko-project/executor:v1.9.0-debug`
 
 **Triage Notes for the Maintainers**
 

 
 | **Description** | **Yes/No** |
 |----------------|---------------|
 | Please check if this a new feature you are proposing        | <ul><li>- [ ] </li></ul>|
 | Please check if the build works in docker but not in kaniko | <ul><li>- [ ] </li></ul>| 
 | Please check if this error is seen when you use `--cache` flag | <ul><li>- [x] </li></ul>|
 | Please check if your dockerfile is a multistage dockerfile | <ul><li>- [ ] </li></ul>|

**Comment by aaron-prindle** [comment]
@stawr93 does the information @tspearconquest provided help clarify things? Is there still discussion needed or a possible kaniko bug here given this info?  Currently I believe this is working as intended IIUC

**Comment by stawr93** [comment]
yeah, @tspearconquest 's answer clarified. It might have been my mistake. I think the issue can be closed/resolved.

In the end I decided to use the registry cache as it is the most simple way to share intermediate build artifacts between pipeline runs.

**Comment by Chococrok** [comment]
Hello, I think the issue is still valid. I used the warmer to populate the cache then used:
```
docker run --rm -v $PWD:/workspace -it gcr.io/kaniko-project/executor:v1.19.2-debug --context "/workspace" --dockerfile "/workspace/Dockerfile" --tar-path /workspace/result.tar --destination=test  --cache=true --cache-dir "/workspace/cache"
```
But kaniko is still complaining that I should use `--cache-repo`

**Comment by Chococrok** [comment]
This update fix the thing:
```
// cacheFlagsValid makes sure the flags passed in related to caching are valid
func cacheFlagsValid() error {
	if !opts.Cache {
		return nil
	}
	// If --cache=true and --no-push=true, then cache repo must be provided
	// since cache can't be inferred from destination
	if opts.CacheRepo == "" && opts.CacheDir == "" && opts.NoPush {
		return errors.New("if using cache with --no-push, specify cache repo with --cache-repo or cache dir with --cache-dir")
	}
	return nil
}
```

**Comment by reitzig** [comment]
FWIW: For creating reproducible MWEs for issues such as #1246, an "offline" way to combine `--no-push` with `--cache` would be very helpful.  In my mind, tying the MWE to a repository makes it both less minimal and less reproducible. 

Actually, I for one don't even know which `--cache-repo` to set that both me and maintainers would be able to use without potentially introducing further problems. 🤷

**Comment by oliverpool** [comment]
I have the same situation as above: I want to build an image without pushing, while using the cache-dir (which I populated using `warmer` in a previous step).

I think the [fix suggested](https://github.com/GoogleContainerTools/kaniko/issues/2456#issuecomment-1875788541) by @Chococrok should be improved, since by default `opts.CacheDir` contains the `/cache` value.

I would suggest checking the existence of the cache-dir in `cacheFlagsValid`

```go
	// If --cache=true and --no-push=true, then cache repo must be provided
	// since cache can't be inferred from destination
	if opts.CacheRepo == "" && opts.NoPush {
		if opts.CacheDir == "" {
			return errors.New("if using cache with --no-push, specify cache repo with --cache-repo or an existing cache dir with --cache-dir")
		} else {
			// check the existence of --cache-dir (since it may contain a default value)
			_, err := os.Stat(opts.CacheDir)
			if err != nil {
				return fmt.Errorf("if using cache with --no-push, specify cache repo with --cache-repo or an existing cache dir with --cache-dir: (%s: %w)", opts.CacheDir, err)
			}
		}
	}
```

instead of

https://github.com/GoogleContainerTools/kaniko/blob/16c43bb97312d0f08d99e0ee552146da09ac0886/cmd/executor/cmd/root.go#L343-L353
